### DISCUSSION TITLE
Advice needed on way forward HW/SW/Storage combination

### DISCUSSION THREAD

**Discussion by tyfoon** [blakeblackshear/frigate#4147] [General]
I’m now using blue iris with 2x4MP and 4x 8MP cams on a dedicated i7 windows PC. Working fine but thinking of changing this to Frigate with the objective to get person/face recognition, better integration in HASS and lastly to save energy. Would like to stay with HASSOS as this has been the most stable vs supervised setups.

After reading many many posts still not sure about the best way forward when it comes to setup in context of HW & Storage.

I’m now running HASOSS on an intel NUC 6cayh (Celeron 3455) with a 128gb m.2 ssd. I’m getting my Coral next week. I don’t use a NAS (and prefer not to) and as far as I understand I can’t add an extra drive in HASSOS. My main questions:

With 6 cams (2x4MP, 4x 8MP): Can this run well together with HASS?

- If so, storage seems my biggest concern; I want to store few weeks of clips which is not possible on 128GB.
- In that case can I best a) replace my m.2 SSD with a 2TB one and then run all on the existing NUC? or b) get another dedicated NUC (eg a J4125 based machine) with a decent 2TB 2.5” drive that only runs frigate and then integrates with the frigate proxy on my existing NUC (basically not changing anything on my current HASSOS machine) .
- any other proposed scenario

Money is not the biggest issue. Just want the best stable, simple and energy efficient setup.

Any other recommendations are welcome.

**Comment by blakeblackshear** [comment]
The resolution of the full size streams doesn't matter much with Frigate since they are just copied to disk. The processor intensive part is related to the stream you choose for the detect role. Typically, that would be a substream. I recommend reading this section of the docs for context: https://docs.frigate.video/guides/camera_setup

If your cameras offer substreams at resolutions and frame rates that meet your needs, you should be totally fine. If not, the processor load goes up dramatically if doing analysis on the full size streams.

**Comment by tyfoon** [comment]
Thanks for quick reply.

My major concern/question however was on the storage part, less on the
detection.

I prefer to store 24/7 plus about a month of clips. Generally this is
between 1 and 2tb of storage usage.

Wondering what the best way forward is based in my info provided in my
first post.

If you have any suggestions, love to hear.



Op zo 23 okt. 2022 12:28 schreef Blake Blackshear ***@***.***

**Comment by blakeblackshear** [comment]
From a storage standpoint either of those options is fine. Running frigate on its own machine will give you more control and flexibility in the long run. Sometimes HassOS makes changes that conflict with Frigate.

**Reply by tyfoon** [reply]
When I go down the dedicated HW route, quick check assuming 6 cams (2x4mp, 4x8mp). For detection low res stream, combined with long term storage for higher res material. Would run Debian on bare metal + docker.

- Would a J4125 NUC (or something similar) + 8gb + USB Coral be enough? (per your HW suggestions on your website)
- Would you recommend a separate disk for OS+Frigate (eg a m2 128gb SSD) + a 2TB 2.5" HDD? Can I set up frigate that way?

Thanks again

**Reply by blakeblackshear** [reply]
You should be able to handle 6 cams assuming you can set 720p and 5fps on the cameras for substreams. You may even be able to make a few of those 1080p if needed for a couple cameras. You can do the disks however you want. Docker volume mounts are very flexible.
